tests/test_broadcasting.py

```diff
@@ -41,14 +41,14 @@
 
     kernels.Linear,
     kernels.Polynomial,
-    kernels.ArcCosine,
-    kernels.Periodic,
+    # kernels.ArcCosine,
+    # kernels.Periodic,
 ]
 
 def _test_no_active_dims(Kern, sess):
     S, N, M, D = 5, 4, 3, 2
     X1 = tf.identity(np.random.randn(S, N, D))
-    X2 = tf.identity(np.random.randn(S, M, D))
+    X2 = tf.identity(np.random.randn(M, D))
```

Comment:
why not add this as an extra test instead of replacing the previous one (which tests slightly different behaviour)?

Reply:
Yes this way is better. X1 and X2 shouldn't have to match leading dims - that case doesn't come up in gpflow at the moment (it would be for multiple datasets/inducing points and multiple observations, i.e. multiple parallel models, which we never do)

Reply:
Yes, the shape now would be `[S, N, S, M]`. You can think of `kern.k(., .)` as a tensordot operation on the last dimension, i.e. reducing the last dimension (here `D`) and then concatenating both tensors from left to right.